File: 2348-Number-of-Zero-Filled-Subarrays/2348-Number-of-Zero-Filled-Subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long ans = 0;
        long n = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0){
                n++;
            } else {
                ans += ( n * (n+1) )/2;
                n = 0;
            }
        }
        if(n != 0 ){
            ans += ( n * (n+1) )/2;

        }
        return ans;   
    }
}

File: 2348-Number-of-Zero-Filled-Subarrays/2348-Number-of-Zero-Filled-Subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        ans = 0
        n = 0 
        nums = [1] + nums
        while len(nums) > 0:
            element = nums.pop()
            if element == 0:
                n += 1
            else:
                ans += (n * (n+1) )/2
                n = 0
            
        return int(ans)